front/lib/api/assistant/reaction.ts

```diff
@@ -0,0 +1,82 @@
+import { Authenticator } from "@app/lib/auth";
+import { ModelId } from "@app/lib/databases";
+import { Message, MessageReaction } from "@app/lib/models";
+import { MessageReactionType } from "@app/types/assistant/conversation";
+
+export async function addOrRemoveMessageReaction(
+  auth: Authenticator,
+  {
+    messageId,
+    userId,
```

Comment:
Let's take a UserType here

Reply:
If it can't be `null` for now then let's not pass userContextUsername/FullName for now

Reply:
This is going to be painfull for when it can be null. I should probably just allow it to be null already?

Reply:
Sure!

In that case to be consistent it's `context : { username, fullName }` caller is responsible to fill it I presume even if user is passed?

Reply:
Yes sir!